### DISCUSSION TITLE
Use this Node-Red flows in Home Assistant and Problems...

### DISCUSSION THREAD

**Discussion by pit65535** [bastelbert/tesla_charging_control_solar_production#5] [Q&A]
Hi, my name is Peter, i have a tesla model3 and a gen3 Wallbox.
I use Home Assistant for automating my home with Node Red installed as Pugin and also teslamate and Grafana on it. Unfortunately I am not very experienced with Node Red. I imported 01_automatic_flow.json and replaced the three Loxone Nodes by Nodes getting that data from my Home Assistant installation.
If I open the Function Node it shows 50 Problems all like this: Cannot find name 'should_charge'.(2304).
What is missing or what am I doing wrong?
I would be very happy, if somebody can help me with that question, as I could not figure out by myself within two days....

**Comment by bastelbert** [comment]
Hi Peter, if you are already using Teslamate, then try Tesla Solar Charger, it’s much more sophisticated than my solution: https://github.com/pkuehnel/TeslaSolarCharger
